### ISSUE TITLE
Is freezing whisper decoder necessary during whisper-flamingo training?

### ISSUE THREAD

**Issue by yakumo221** [roudimit/whisper-flamingo#12]
Thank you for releasing the code accompanying your excellent work: Whisper-Flamingo.

In Figure 1 of the paper, it is stated that during the training of Whisper-Flamingo, the decoder of the fine-tuned Whisper model is frozen, and only the gated cross-attention layers and a linear projection are trained.

However, after reviewing the released codebase, I could not find any code that explicitly freezes the parameters of the Whisper decoder. Whether the decoder was actually frozen during training in the experiments?

**Comment by roudimit** [comment]
Thanks for your interest! Yes, the decoder was frozen and only the gated cross attention layers in the decoder were trained. This is controlled by the [optimizer](https://github.com/roudimit/whisper-flamingo/blob/6ea6b48f325b8f3c95eabb8e7c03d00304276a16/whisper_ft_muavic_video.py#L266). [It sets](https://github.com/roudimit/whisper-flamingo/blob/6ea6b48f325b8f3c95eabb8e7c03d00304276a16/utils.py#L439) only the parameters from the gated cross attention layers to be trainable.

For [mWhisper-Flamingo](https://arxiv.org/abs/2502.01547) we also set the AV-HuBERT video encoder to be [trainable](https://github.com/roudimit/whisper-flamingo/blob/6ea6b48f325b8f3c95eabb8e7c03d00304276a16/config/audio-visual/av_multi_medium.yaml#L26) (ie. fine-tuned).

**Comment by yakumo221** [comment]
I see. That's awesome! Thanks a lot!!

**Comment by yakumo221** [comment]
Another question: when specifying optimization parameters in transformers.optimization (e.g. AdamW), will the parameters that are not included in the optimizer still compute their gradients and occupy GPU memory during training?

**Comment by roudimit** [comment]
[Yes](https://www.reddit.com/r/learnmachinelearning/comments/153yywo/how_does_gradient_calculation_work_when_layers_in/)

**Comment by yakumo221** [comment]
get it
